pkg/pgmodel/migrate.go

```diff
@@ -26,133 +28,344 @@ const (
+			fullName := filepath.Join(dirName, fileName)
+			file, err = migrations.MigrationFiles.Open(fullName)
+			if err != nil {
+				return fmt.Errorf("unable to get migration script from toc: name %s, err %w", fullName, err)
+			}
+
+			if stat, err = file.Stat(); err != nil {
+				return fmt.Errorf("unable to stat migration script from toc: name %s, err %w", fullName, err)
+			}
+
+			// Ignoring directories.
+			if stat.IsDir() {
+				continue
+			}
+
+			entries = append(entries, fileName)
+		}
+	} else {
+		// Otherwise, order the files by their numeric prefix, delimited by `-` (if one exists).
+		fileEntries, err := f.Readdir(-1)
+		if err != nil {
+			return fmt.Errorf("unable to read migration scripts directory: name %s, err %w", dirName, err)
+		}
+
+		entries = orderFilesNaturally(fileEntries)
+	}
+
+	for _, e := range entries {
+		fileName := filepath.Join(dirName, e)
+		f, err := migrations.MigrationFiles.Open(fileName)
 		if err != nil {
-			return
+			return fmt.Errorf("unable to get migration script: name %s, err %w", fileName, err)
 		}
-		if sourceErr != nil {
-			err = sourceErr
-			return
+		contents, err := replaceSchemaNames(f)
+		if err != nil {
+			return fmt.Errorf("unable to read migration script: name %s, err %w", fileName, err)
 		}
-		if databaseErr != nil {
-			err = databaseErr
-			return
+		_, err = tx.Exec(context.Background(), string(contents))
+		if err != nil {
+			return fmt.Errorf("error executing migration script: name %s, err %w", fileName, err)
 		}
-	}()
+	}
+
+	return nil
+}
+
+// orderFilesNaturally orders the file names by their numberic prefix, ignoring
+// directories or files which are not formatted correctly.
+func orderFilesNaturally(entries []os.FileInfo) []string {
+	pp := prefixedNames{}
+	var (
+		prefix int
+		name   string
+	)
+
+	for _, entry := range entries {
+		// Ignoring directories.
+		if entry.IsDir() {
+			continue
+		}
+
+		_, err := fmt.Sscanf(entry.Name(), "%d-%s", &prefix, &name)
+		// Ignore malformated file names.
+		if err != nil {
+			continue
+		}
+		pp = append(pp, prefixedName{prefix, entry.Name()})
+	}
+
+	sort.Sort(pp)
+	return pp.getNames()
+}
 
-	err = m.Up()
-	//ignore no change errors as we want this idempotent. Being up to date is not a bad thing.
-	if err == migrate.ErrNoChange {
-		err = nil
+func replaceSchemaNames(r io.ReadCloser) (string, error) {
+	buf := new(bytes.Buffer)
+	_, err := buf.ReadFrom(r)
+	if err != nil {
+		return "", err
 	}
+	err = r.Close()
 	if err != nil {
-		return err
+		return "", err
 	}
+	s := buf.String()
+	s = strings.ReplaceAll(s, "SCHEMA_CATALOG", catalogSchema)
+	s = strings.ReplaceAll(s, "SCHEMA_EXT", extSchema)
+	s = strings.ReplaceAll(s, "SCHEMA_PROM", promSchema)
+	s = strings.ReplaceAll(s, "SCHEMA_SERIES", seriesViewSchema)
+	s = strings.ReplaceAll(s, "SCHEMA_METRIC", metricViewSchema)
+	s = strings.ReplaceAll(s, "SCHEMA_DATA", dataSchema)
+	s = strings.ReplaceAll(s, "SCHEMA_DATA_SERIES", dataSeriesSchema)
```

Comment:
we should switch these two lines; it won't work if [done in the current order](https://play.golang.org/p/h7HeUFqqi6U)

Reply:
we should probably use a [strings.Replacer](https://golang.org/pkg/strings/#Replacer) for performance, though it won't fix the bug